src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -438,53 +503,130 @@ void GfxBatchHbaoTest::testFlippedHBAOData(const TestDataType& data,
   auto&& data = TestData[testCaseInstanceId()];
-  setTestCaseDescription(
-      Cr::Utility::formatString("{}, perspective", data.name));
+  setTestCaseDescription(Cr::Utility::format("{}, perspective", data.name));
   testHBAOData(
-      data,
-      Cr::Utility::formatString("{}.{}.png", baseTestFilename, data.filename),
+      data, Cr::Utility::format("{}.{}.png", baseTestFilename, data.filename),
       perspectiveData);
 
 }  // GfxBatchHbaoTest::testPerspective()
 
 void GfxBatchHbaoTest::testOrthographic() {
   auto&& data = TestData[testCaseInstanceId()];
-  setTestCaseDescription(
-      Cr::Utility::formatString("{}, orthographic", data.name));
-  testHBAOData(data,
-               Cr::Utility::formatString("{}.{}-ortho.png", baseTestFilename,
-                                         data.filename),
-               orthographicData);
+  setTestCaseDescription(Cr::Utility::format("{}, orthographic", data.name));
+  testHBAOData(
+      data,
+      Cr::Utility::format("{}.{}-ortho.png", baseTestFilename, data.filename),
+      orthographicData);
 
 }  // GfxBatchHbaoTest::testOrthographic()
 
 void GfxBatchHbaoTest::testPerspectiveFlipped() {
   auto&& data = TestData[testCaseInstanceId()];
   setTestCaseDescription(
-      Cr::Utility::formatString("{}, perspective, flipped", data.name));
+      Cr::Utility::format("{}, perspective, flipped", data.name));
   testFlippedHBAOData(
-      data,
-      Cr::Utility::formatString("{}.{}.png", baseTestFilename, data.filename),
+      data, Cr::Utility::format("{}.{}.png", baseTestFilename, data.filename),
       flippedPerspectiveData);
 
 }  // GfxBatchHbaoTest::testPerspective()
 
-// void GfxBatchHbaoTest::benchmarkPerspective() {
-//   auto&& data = TestData[testCaseInstanceId()];
-//   setTestCaseDescription(
-//       Cr::Utility::formatString("{}, perspective benchmark.", data.name));
+void GfxBatchHbaoTest::benchmarkHBAOData(const TestDataType& data,
+                                         Mn::Matrix4 projMatrix) {
+  if ((data.config.flags() & esp::gfx_batch::HbaoFlag::LayeredImageLoadStore) &&
+      !(
+#ifdef MAGNUM_TARGET_GLES
+          Mn::GL::Context::current().isVersionSupported(
+              Mn::GL::Version::GLES310)
+#else
+          Mn::GL::Context::current()
+              .isExtensionSupported<
+                  Mn::GL::Extensions::ARB::shader_image_load_store>()
+#endif
+              ))
+    CORRADE_SKIP("Image load/store not supported");
+
+  // Scale image size to use for benchmark uniformly so projection matrix aspect
+  // ration not affected.
+  Mn::Vector2i BenchImageSize = Size * 4;
+  const size_t arraySize = BenchImageSize.x() * BenchImageSize.y() * 8;
+
+  // For benchmarks source color and depth are just empty white images
+
+  Cr::Containers::Array<char> depthData{Cr::DirectInit, arraySize, '\xff'};
+  Mn::Trade::ImageData2D depth{Mn::PixelFormat::Depth32F, BenchImageSize,
+                               std::move(depthData)};
```

Comment:
If you use a view, you don't need the move:

```suggestion
  Mn::Trade::ImageDataView2D depth{Mn::PixelFormat::Depth32F, BenchImageSize,
                               depthData};
```

I often even make the image view temporary in the `setSubImage()` call, not sure what clang-format makes out of that tho.